src/app/WirelessIno/app/src/main/java/se/sics/sse/fresta/wirelessino/Main.java

```diff
@@ -5,117 +5,280 @@
+import android.widget.EditText;
+import android.widget.SeekBar;
+import android.widget.Toast;
+
+/**
+ * This is the Main class which is the main activity for the application.
+ * All the interactive GUI elements are defined here as well as their
+ * corresponding methods and calculations.
+ */
 
 public class Main extends Activity {
-	public static final String TAG = "WirelessIno";
-	public static Socket socket = null;
-	public static final boolean D = true; // The debug option
-	
-	private static PrintWriter out = null;
-	private PadView view = null;
-	private Menu menu = null;
-	
-	private static final int EXIT_INDEX = 0;		// Menu bar: exit  
-	private static final int DISCONNECT_INDEX = 1;	// Menu bar: disconnect
-	private static final int CONFIG_INDEX = 2	;	// Menu bar: WiFi configuration
 
-	protected void onCreate(Bundle savedInstanceState) {
-		super.onCreate(savedInstanceState);
-		
-		view = new PadView(this);
-		setContentView(view);
-	}
-	
-	/* 
-	 * Add a disconnect option when connected to a socket. 
-	 */
-	protected void onResume() {
-		/* Disable the disconnect option if no connection has been established */
-		updateMenuVisibility();
-		super.onResume();
-	}
-	
-	/* 
-	 * Add menu options 
-	 */
-	public boolean onCreateOptionsMenu(Menu menu) {
-		this.menu = menu;
+
+
+
+    private SeekBar topBar;
+    private SeekBar bottomBar;
+    private Button modeButton;
+    private boolean isACCenabled;
+
+
+    public static final String TAG = "WirelessIno";
+    public static Socket socket = null;
+    public static final boolean D = true; // The debug option
+
+    private static PrintWriter out = null;
+    private Menu menu = null;
+
+    private static final int EXIT_INDEX = 0;        // Menu bar: exit
+    private static final int DISCONNECT_INDEX = 1;    // Menu bar: disconnect
+    private static final int CONFIG_INDEX = 2;    // Menu bar: WiFi configuration
+
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        topBar = (SeekBar) findViewById(R.id.TopBar);
+        bottomBar = (SeekBar) findViewById(R.id.BottomBar);
+
+        /*Listener for change in SeekBars (Sliders) */
+
+        /**
+     * Listener for topBar
+     */
+        topBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
+
+            @Override
+            public void onStopTrackingTouch(SeekBar seekBar) {
+                /* The value 100 turns into "0" in calculations */
+                topBar.setProgress(100);
+            }
+
+            @Override
+            public void onStartTrackingTouch(SeekBar seekBar) {
+                // TODO Auto-generated method stub
+            }
+
+            @Override
+            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
+                transformPWM();
+            }
+        });
+
+        /**
+         * Listener for bottomBar

```

Comment:
A better naming convention for top and bottom bars would be great. To further explain which one is what.